ext_mongodb.php

```diff
@@ -715,17 +720,43 @@ function __debugInfo() : array;
 {
 	use DenySerialization;
 
-	public function __construct(private string $code, private ?mixed $scope = NULL)
+	private $code;
+
+	public function __construct(string $code, private ?mixed $scope = NULL)
 	{
+		if ( strstr( $code, "\0" ) !== false )
+		{
+			throw new \MongoDB\Driver\Exception\InvalidArgumentException( "Code cannot contain null bytes" );
+		}
+		$this->code = $code;
 	}
 
 	public function __debugInfo() : array
 	{
 		return [
-			'javascript' => $this->code,
+			'code' => $this->code,
 			'scope' => (object) $this->scope
 		];
 	}
+
+	public function getCode() : string
+	{
+		return $this->code;
+	}
+
+	public function getScope() : mixed
+	{
+		if ( $this->scope !== NULL )
+		{
+			return (object) $this->scope;
```

Comment:
I wasn't insinuating that HHVM was round-tripping anything; rather, that this was just to pass the test for PHPC, which round-trips. Find to leave this as-is. There is a subtle behavior difference if we deal with nested documents but I think it's unlikely that users will encounter it or even care.